lib/shopify_api/application.ex

```diff
@@ -1,15 +1,18 @@
 defmodule ShopifyAPI.Application do
   use Application
 
+  alias ShopifyAPI.AvailabilityTracker
+
   # See https://hexdocs.pm/elixir/Application.html
   # for more information on OTP Applications
   def start(_type, _args) do
+    AvailabilityTracker.init()
```

Comment:
Doesn't this make more sense to exist as a `gen_server` hooked up to the rest of the supervision tree?

Reply:
What exactly would the GenServer be doing here? The `:ets` table is public and not tied to any particular process. The `:ets` table itself is not supervised, but I am not sure how to do that without making it private to a process. I think making the `:ets` table private would introduce a bottleneck of the GenServer itself. 
This actually behaves the same way that the previous implementation in terms of supervision (except for in the old implementation the GenServer would not be able to be restarted by the supervisor as it would try to init the `:ets` table again and fail (luckily it was never actually used so it was not likely to fail)).

Reply:
I suppose there could be a worker pool that accessed the `:ets` table, but I think that may require a fair amount more work and I don't know if we expect the `:ets` table to crash.

Reply:
I think it might just be me being overcautious - I'm mostly worried about what making the supervisor process the owner of the table does in scenarios where crashes occur in supervised processes. I've typically seen this instead implemented as a `GenServer` process that owns the table and handles single-process writes, while other processes can read from the table in parallel.

In this case, I don't think there's anything we can really do to crash the table proper, so it's _probably_ OK?